plugins/postcss-logical-viewport-units/src/index.ts

```diff
@@ -0,0 +1,97 @@
+import type { PluginCreator } from 'postcss';
+import { hasSupportsAtRuleAncestor } from './has-supports-at-rule-ancestor';
+import { transform } from './transform';
+
+/** postcss-logical-viewport-units plugin options */
+export type pluginOptions = {
+	/** Preserve the original notation. default: false */
+	preserve?: boolean,
+	/** Control how logical viewport units are replaced. default: "horizontal" */
+	writingMode: 'horizontal' | 'vertical',
+};
+
+const creator: PluginCreator<pluginOptions> = (opts?: pluginOptions) => {
+	const options = Object.assign(
+		// Default options
+		{
+			writingMode: 'horizontal',
+			preserve: true,
+		},
+		// Provided options
+		opts,
+	);
+
+	if (options.writingMode !== 'vertical') {
+		options.writingMode = 'horizontal';
+	}
+
+	const replacements: { vi: 'vw' | 'vh', vb: 'vw' | 'vh' } = {
+		vb: 'vh',
+		vi: 'vw',
+	};
+
+	if (options.writingMode === 'vertical') {
+		replacements.vb = 'vw';
+		replacements.vi = 'vh';
+	}
+
+	return {
+		postcssPlugin: 'postcss-logical-viewport-units',
+		Declaration(decl, { atRule }) {
+			{
+				// Fast check
+				const lowerCaseValue = decl.value.toLowerCase();
+				if (!(lowerCaseValue.includes('vb') || lowerCaseValue.includes('vi'))) {
+					return;
+				}
+
+				// Declaration already has a fallback
+				const prev = decl.prev();
+				if (prev && prev.type === 'decl' && prev.prop === decl.prop) {
+					return;
+				}
```

Comment:
Seeing this makes me wonder a couple of things:

Should we have some global util for this? We have this in the form of checking from the parent and also checking previous. 
Should it be standard?
Could there be any use case in which you (as a user) wouldn't want this behaviour?

Reply:
> Could there be any use case in which you (as a user) wouldn't want this behaviour?

I don't think this will be observable to users.
Either they have added a manual fallback and it will work or the plugins will add the needed fallbacks automatically.

The only thing that will be omitted are some unneeded intermediate values.

I am indeed being extra cautious with this plugin in particular because it is for a unit.
With multiple plugins generating fallbacks it can quickly escalate because a single declaration can contain multiple features.

plugin 1 : `alpha` -> `one`
plugin 2: `beta` -> `two`

```css
.foo {
  bar: alpha beta;
}
```

plugin 1 results in :

```css
.foo {
  bar: alpha beta;
  bar: one beta;
}
```

plugin 2 results in :

```css
.foo {
  modern: alpha beta;
  modern: alpha two;
  modern: one beta;
  modern: one two;
}
```

plugin 1 runs again :

```css
.foo {
  modern: alpha beta;
  modern: alpha two;
  modern: one two;
  modern: one beta;
  modern: one two;
}
```

------

By short circuiting we get :

plugin 1 results in :

```css
.foo {
  bar: alpha beta;
  bar: one beta;
}
```

plugin 2 results in :

```css
.foo {
  modern: alpha beta;
  modern: one beta;
  modern: one two;
}
```

Ideally we wouldn't have `one beta` but that is the cost of having separate plugins and not being lightningcss :)

--------

> Should it be standard?

Maybe it should.
Wouldn't be complex to add.

--------

> Should we have some global util for this

Yes, but I don't like the progressive custom properties thing I made.
It is still a pain to setup and it makes things more complex, not less so.

So it can be a shared thing but it must be a real reduction in complexity :)

Reply:
I also already documented this :)

https://github.com/csstools/postcss-plugins/wiki/Plugin-best-practices#manual-fallback-declarations